src/view/domconverter.js

```diff
@@ -371,7 +371,15 @@ export default class DomConverter {
 	 */
 	domToView( domNode, options = {} ) {
 		if ( isBlockFiller( domNode, this.blockFiller ) ) {
-			return null;
+			const isSingle = domNode.parentNode && domNode.parentNode.childNodes.length <= 1;
```

Comment:
I think we should have this entire logic in `isBlockFiller()`. This is exactly what it's supposed to check.